### ISSUE TITLE
Apollo 3 customScalarsMapping for ID

### ISSUE THREAD

**Issue by Saecki** [apollographql/apollo-kotlin#3748]
After upgrading to apollo 3, and replacing `customTypeMapping` with `customScalarsMapping`, mapping the `ID` scalar to `kotlin.Long` will throw this error: 
```
Apollo: unknown custom scalar(s) in customScalarsMapping: ID
```
Is there another way to map standard scalars now, or is this usecase simply not supported?

**Comment by martinbonnin** [comment]
Good catch. Indeed, this hasn't made it to 3.x. The overall reason is that `ID` isn't technically a custom scalar since it's builtin and should always be serializable to a `String`. From [the spec](https://spec.graphql.org/draft/#sec-ID):

```
The ID scalar type represents a unique identifier, often used to refetch an object or as the key for a cache. 
The ID type is serialized in the same way as a String; however, it is not intended to be human-readable. 
While it is often numeric, it should always serialize as a String.
```

I'm curious what the use case is. Do you need to do some maths on the IDs? Or is it because storing a Long is more compact than a String? Or something else?

**Comment by Saecki** [comment]
Yeah, peace of mind. It probably doesn't matter but I like the thought of `ID`s not being heap allocated and fixed size.
Also comparing elements by an `ID` of a numeric type should be much faster than comparing strings.

**Comment by martinbonnin** [comment]
👍 That's fair. We'll look into adding this back. 

If we're doing this, I think `ID` shouldn't be a special case but it should be possible to map any scalar (builtin or custom):

```kotlin
/**
 *
 * @param graphqlName: the name of the scalar in the schema
 * @param kotlinName: fully qualified name of the kotlin type to use in models
 * @param adapterName: fully qualified name of the adapter used to adapt the Graph type to the Kotlin type
 * The adapter must have a no-arg constructor or be an `object` singleton
 */
fun Service.mapScalar(graphqlName: String, kotlinName: String, adapterName: String?)
```

That'd make it possible to avoid the lookups in the `CustomScalarAdapters` at runtime:

```kotlin
apollo {
  mapScalar("ID", "kotlin.Long", "com.apollographql.apollo3.api.LongAdapter")
}
```

We could even think of nice overrides for the (Apollo) builtin adapters:

```kotlin
apollo {
  mapScalarToLong("ID")
  mapScalarToMap("Json")
  ...
}
```

**Comment by Saecki** [comment]
That looks great!

**Comment by simon-the-canadian** [comment]
I'm in the process of migrating from v2 to v3 and I would really appreciate the support for having a custom type converter for the `ID` scalar like v2 allowed.

**Comment by martinbonnin** [comment]
Fixed in 3.1.0. You can now use `mapScalar()` for any scalar in your Gradle configuration
